JIT optimization converts long value to integer

### Expected Behavior

Long values in `application.yml` shouldn't be converted to **Integer** if configuration expects **Long**
```yaml
micronaut:
  server:
    max-request-size: 2147483648 # 2gb (1024 * 1024 * 1024 * 2)
    multipart:
      enabled: true
      max-file-size: 2147483648 # 2gb (1024 * 1024 * 1024 * 2)
```



### Actual Behaviour


```
[~/www/project] (master) $ ./gradlew optimizedRun

> Task :compileJava
Note: Generating OpenAPI Documentation
Note: Writing OpenAPI YAML to destination: /home/message/www/project/build/classes/java/main/META-INF/swagger/project-0.1.yml
Note: Writing reflect-config.json file to destination: META-INF/native-image/de.project/project/reflect-config.json
Note: Creating bean classes for 59 type elements

> Task :prepareJitOptimizations
2021-12-21 15:19:51,453 [main] INFO  io.micronaut.aot.MicronautAotOptimizer$Runner:312 - Detected environments: []
2021-12-21 15:19:51,553 [main] INFO  io.micronaut.aot.std.sourcegen.YamlPropertySourceGenerator:65 - Converting application.yml into Java based configuration
2021-12-21 15:19:51,566 [ForkJoinPool.commonPool-worker-11] INFO  io.micronaut.aot.std.sourcegen.YamlPropertySourceGenerator:65 - Converting application.yml into Java based configuration
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
        at io.micronaut.aot.cli.Main.executeInIsolatedLoader(Main.java:118)
        at io.micronaut.aot.cli.Main.run(Main.java:92)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1939)
        at picocli.CommandLine.access$1300(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2311)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
        at picocli.CommandLine.execute(CommandLine.java:2078)
        at io.micronaut.aot.cli.Main.execute(Main.java:139)
        at io.micronaut.aot.cli.Main.main(Main.java:143)
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.micronaut.aot.cli.Main.executeInIsolatedLoader(Main.java:112)
        ... 10 more
Caused by: java.lang.RuntimeException: Compilation errors:
File ApplicationStaticPropertySource.java, line: 13, integer number too large
File ApplicationStaticPropertySource.java, line: 17, integer number too large

        at io.micronaut.aot.MicronautAotOptimizer.throwCompilationError(MicronautAotOptimizer.java:229)
        at io.micronaut.aot.MicronautAotOptimizer.compileGeneratedSources(MicronautAotOptimizer.java:135)
        at io.micronaut.aot.MicronautAotOptimizer.access$200(MicronautAotOptimizer.java:95)
        at io.micronaut.aot.MicronautAotOptimizer$Runner.execute(MicronautAotOptimizer.java:319)
        at io.micronaut.aot.MicronautAotOptimizer.execute(MicronautAotOptimizer.java:191)
        ... 15 more

> Task :prepareJitOptimizations FAILED
```

### Steps To Reproduce

_No response_

### Environment Information

- Ubuntu 20 LTS

### Example Application

_No response_

### Version

3.2.3

--- a/aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy
+++ b/aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy
@@ -1,7 +1,7 @@
 /*
  * Copyright 2017-2021 original authors
  *
- * Licensed under the Apache License, Version 2.0 (the "License");
+ * Licensed under the Apache License, Version 2.0 (the "License")
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
@@ -27,8 +27,16 @@ import io.micronaut.context.ApplicationContextBuilder
 import spock.lang.Specification
 import spock.lang.TempDir
 
+import javax.tools.Diagnostic
+import javax.tools.DiagnosticCollector
+import javax.tools.JavaCompiler
+import javax.tools.JavaFileObject
+import javax.tools.StandardJavaFileManager
+import javax.tools.ToolProvider
+import java.nio.file.Files
 import java.nio.file.Path
 import java.util.function.Function
+import java.util.stream.Collectors
 
 @CompileStatic
 abstract class AbstractSourceGeneratorSpec extends Specification {
@@ -194,6 +202,57 @@ abstract class AbstractSourceGeneratorSpec extends Specification {
             sources.keySet().collect { key ->
                 "${key.packageName}.${key.typeSpec.name}".toString()
             } as Set<String>
+        }
+
+        void compiles(List<File> classpath = []) {
+            def compiler = ToolProvider.getSystemJavaCompiler()
+            def ds = new DiagnosticCollector<>()
+            try (def mgr = compiler.getStandardFileManager(ds, null, null)) {
+                def fullClasspath = System.getProperty("java.class.path").split(File.pathSeparator).collect {
+                    new File(it)
+                }.findAll { it.name.endsWith(".jar") } + classpath
+                def sourceDir = testDirectory.resolve("sources").toFile()
+                def output = testDirectory.resolve("compiled").toFile()
+                List<String> options = compilerOptions(output, fullClasspath)
+                sources.keySet().each {
+                    it.writeTo(sourceDir)
+                }
+                def filesToCompile = Files.walk(sourceDir.toPath()).filter {
+                    it.toFile().isFile() && it.toFile().name.endsWith(".java")
+                }.map {
+                    it.toFile()
+                }.collect(Collectors.toList())
+                if (output.mkdirs()) {
+                    def sources = mgr.getJavaFileObjectsFromFiles(filesToCompile)
+                    def task = compiler.getTask(null, mgr, ds, options, null, sources)
+                    task.call()
+                }
+            } catch (IOException e) {
+                throw new RuntimeException("Unable to compile generated classes", e)
+            }
+            def diagnostics = ds.diagnostics
+                    .findAll { d -> d.kind == Diagnostic.Kind.ERROR }
+            if (!diagnostics.isEmpty()) {
+                diagnostics.each {
+                    println "ERROR: ${((Diagnostic)it).getMessage(Locale.ENGLISH)}"
+                }
+                throw new AssertionError("Expected sources to compile but they didn't")
+            }
+        }
+
+        private static List<String> compilerOptions(File dstDir,
+                                                    List<File> classPath) {
+            def options = new ArrayList<String>()
+            options.add("-source")
+            options.add("1.8")
+            options.add("-target")
+            options.add("1.8")
+            options.add("-classpath")
+            String cp = classPath.collect { it.absolutePath }.join(File.pathSeparator)
+            options.add(cp)
+            options.add("-d")
+            options.add(dstDir.getAbsolutePath())
+            options
         }
     }
 

--- a/aot-std-optimizers/src/main/java/io/micronaut/aot/std/sourcegen/MapPropertySourceGenerator.java
+++ b/aot-std-optimizers/src/main/java/io/micronaut/aot/std/sourcegen/MapPropertySourceGenerator.java
@@ -71,12 +71,29 @@ public class MapPropertySourceGenerator extends AbstractSingleClassFileGenerator
             mapBuilder.add("put(\"" + key + "\", ");
             if (value == null) {
                 mapBuilder.add("null");
-            } else if (CharSequence.class.isAssignableFrom(value.getClass())) {
-                mapBuilder.add("\"" + value + "\"");
-            } else if (Number.class.isAssignableFrom(value.getClass()) || Boolean.class.isAssignableFrom(value.getClass())) {
-                mapBuilder.add(String.valueOf(value));
             } else {
-                throw new UnsupportedOperationException("Configuration map contains an entry of type " + value.getClass() + " which is not supported yet. Please file a bug report.");
+                Class<?> valueClass = value.getClass();
+                if (CharSequence.class.isAssignableFrom(valueClass)) {
+                    mapBuilder.add("\"" + value + "\"");
+                } else if (Number.class.isAssignableFrom(valueClass) || Boolean.class.isAssignableFrom(valueClass)) {
+                    String format = String.valueOf(value);
+                    String prefix = "";
+                    String appendix = "";
+                    if (Long.class.equals(valueClass)) {
+                        appendix = "L";
+                    } else if (Double.class.equals(valueClass)) {
+                        appendix = "D";
+                    } else if (Float.class.equals(valueClass)) {
+                        appendix = "F";
+                    } else if (Byte.class.equals(valueClass)) {
+                        prefix = "(byte) ";
+                    } else if (Short.class.equals(valueClass)) {
+                        prefix = "(short) ";
+                    }
+                    mapBuilder.add(prefix + format + appendix);
+                } else {
+                    throw new UnsupportedOperationException("Configuration map contains an entry of type " + valueClass + " which is not supported yet. Please file a bug report.");
+                }
             }
             mapBuilder.add(");\n");
         }

--- a/aot-std-optimizers/src/test/groovy/io/micronaut/aot/std/sourcegen/MapPropertySourceGeneratorTest.groovy
+++ b/aot-std-optimizers/src/test/groovy/io/micronaut/aot/std/sourcegen/MapPropertySourceGeneratorTest.groovy
@@ -156,20 +156,22 @@ public class TestStaticPropertySource extends MapPropertySource {
 }
 """
             }
+            compiles()
         }
 
         where:
-        value         | literal
-        'a string'    | '"a string"'
-        (int) 1       | '1'
-        (short) 1     | '1'
-        (byte) 1      | '1'
-        (long) 1      | '1'
-        2f            | '2.0'
-        2d            | '2.0'
-        true          | true
-        false         | false
-        Boolean.TRUE  | true
-        Boolean.FALSE | false
+        value              | literal
+        'a string'         | '"a string"'
+        (int) 1            | '1'
+        (short) 1          | '(short) 1'
+        (byte) 1           | '(byte) 1'
+        (long) 1           | '1L'
+        (long) 2147483648L | '2147483648L'
+        2f                 | '2.0F'
+        2d                 | '2.0D'
+        true               | true
+        false              | false
+        Boolean.TRUE       | true
+        Boolean.FALSE      | false
     }
 }

